language-design.md

```diff
@@ -16,19 +14,29 @@ are:
+  + I absolutely agree with this statement made by Andreas Stefik, that is quoted in Pavlus' article. Since the first day I have started learning to program, I have felt like most of the programming languages are not user-friendly, or just targeted to be understood by only "technical wizards with magical math powers".
+  +  I certainly had this feeling when learning Haskell. It seemed as though everything about the language was created with a paradigm of category theory which must be understood (or worse, used in ignorance) to understand how to print (since I/O must occur through a Monad).  This is even more true when considering the Haskellian typeclasses, generics, and the type system. A solid understanding of functors, monoids, monads, and applicatives is expected by the designers. While Haskell is an extreme example, many programming languages expect substantial mathematical reasoning or domain knowledge.
++ "A major reason to eradicate these false stereotypes is that they perpetuate biases against women. Evidence shows that a hostile culture contributes to the 'leaky pipeline', the phenomenon of women leaving tech despite having the interest, skills, and education" (Yang and Rabkin, 2015).
+  + It is absolutely true that there are far fewer women than men in the tech industry, despite the increasing numbers of women earning technical degrees. I feel that this statement would have benefited from an elaboration on the evidence indicating that the primary reason for women leave the industry is the gendered perception of programming language hierarchy, as opposed to a more general form of sexism or preconceptions. It is imperative to understand exactly how the gendered perceptions negatively impacting women are formed and perpetuated in order to bring the tech industry closer to an equal opportunity workplace for both men and women.
++ "The development environment suffers from literally decades of accretion of stupid hacks from a community containing, to a first-order approximation, zero software engineers." (Smith, 2016)
+  + This quotation from the additional reading on the R programming language is included because it was very reminiscent of a discussion with Professor Ben, in which we were discussing the separation of domain knowledge and programming language design / software engineering experience.
+  + This highlights the importance of the work we did at the Hive in learning how to interview domain experts. Since these are the individuals with the fluency in the domain, it is essential that software developers creating tools for them are able to empathize with their needs so that the language also has fluency. The other direction is also important (the lack of which inspired this quote about R). When domain specialists are alone in developing a language, the lack of designers and software engineers may lead to unintuitive solutions or software which is difficult to extend.
```

Comment:
I think you raise a very important point in that the community or committee that works on developing an API/PL/DSL must have some balance between Software Engineers and Domain Experts. Even a non-solitary design process can create a problematic language full of major extensibility and logic issues if people don't approach the design from different angles. Now, as Steele points out in his talk/article, there is a balance between "Cathedral-Style" and "Bazaar-Style", but we still need to be concerned with the homogeneity vs. diversity balance in the community behind the language to ensure it suits users needs and can last.